### PR TITLE
[Messenger] Testing LoggingMiddleware and minor improvements

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

- This add tests for `LoggingMiddleware` class 
- Similar to [`translator.logging`](https://github.com/symfony/symfony/blob/6bbb5bcc5294fdd540bec35541f77b87e212fc5f/src/Symfony/Component/Translation/LoggingTranslator.php#L33) a logging service doesn't make sense without a [hard `logger` dependency](https://github.com/symfony/symfony/blob/6bbb5bcc5294fdd540bec35541f77b87e212fc5f/src/Symfony/Bundle/FrameworkBundle/Resources/config/translation.xml#L25-L29) and according to:
https://github.com/symfony/symfony/blob/6bbb5bcc5294fdd540bec35541f77b87e212fc5f/src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php#L52-L54

   it must be removed if `logger` alias doesn't exists.

### CODE REVIEW

File: src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php [hunk 1]
```diff
@@ -49,10 +49,6 @@ public function process(ContainerBuilder $container)
             return;
         }
 
-        if (!$container->getParameter('kernel.debug') || !$container->has('logger')) {
-            $container->removeDefinition('messenger.middleware.debug.logging');
-        }
```
Comment (yceruto) [inline src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php]: I needed move this part to `FrameworkExtension` to make tests pass ([see old failures](https://travis-ci.org/symfony/symfony/jobs/365937393#L2677))

File: src/Symfony/Component/Messenger/composer.json [hunk 2]
```diff
@@ -19,12 +19,12 @@
         "php": "^7.1.3"
     },
     "require-dev": {
+        "psr/log": "~1.0",
         "symfony/serializer": "~3.4|~4.0",
         "symfony/dependency-injection": "~3.4.6|~4.0",
         "symfony/http-kernel": "~3.4|~4.0",
         "symfony/property-access": "~3.4|~4.0",
         "symfony/var-dumper": "~3.4|~4.0",
-        "symfony/property-access": "~3.4|~4.0",
```
Comment (yceruto) [inline src/Symfony/Component/Messenger/composer.json]: duplicated entry (unrelated change)

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 3]
```diff
@@ -1446,6 +1446,10 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
 
         $loader->load('messenger.xml');
 
+        if (!$container->getParameter('kernel.debug') || !$container->hasAlias('logger')) {
```
Comment (ogizanagi) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Actually, the `!$container->hasAlias('logger')` part could be removed, as the Fwb always registers a default logger implementation since #24300

But anyway that'll be handled differently with #26864 as it'll have to be opt-in explicitly.

Comment (sroze) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: AFAIK doing so will always remove the middleware. I don't see any value in changing what was just changed in #26896 :)

File: src/Symfony/Component/Messenger/Tests/Debug/LoggingMiddlewareTest.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        $message = new DummyMessage('Hey');
+
+        $logger = $this->createMock(LoggerInterface::class);
+        $logger->expects($this->exactly(2))->method('debug');
+
+        $next = $this->createPartialMock(\stdClass::class, array('__invoke'));
+        $next->expects($this->once())->method('__invoke')->with($message);
+
+        (new LoggingMiddleware($logger))->handle($message, $next);
```
Comment (sroze) [inline src/Symfony/Component/Messenger/Tests/Debug/LoggingMiddlewareTest.php]: Can you test that it returns the value returned by the previous middleware as well?

**Review by sroze — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 3 above)
Comment (ogizanagi) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: The `!$container->hasAlias('logger')` check in the fwb extension will indeed lead to the middleware removal, even if there is in fact a `logger` alias declared by the monolog bundle or the kernel `LoggerPass`, so it's wrong. But as stated above, this check is not necessary anyway, as there will always be a `logger` alias.

Regarding #26896 , I agree with @yceruto: It doesn't really make sense to me having a LoggingMiddleware without a hard logger dependency. It solves an issue we usually solve at compilation time by removing proper services.

File: src/Symfony/Component/Messenger/Tests/Debug/LoggingMiddlewareTest.php (hunk 4 above)
Comment (yceruto) [reply src/Symfony/Component/Messenger/Tests/Debug/LoggingMiddlewareTest.php]: Done.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 3 above)
Comment (yceruto) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: I'll try this again soon.

**Comment by yceruto** [comment]
@ogizanagi the [tests fail](https://travis-ci.org/symfony/symfony/jobs/366133655#L2687) again, I guess `LoggerPass` is processed after `MessengerPass`?
https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php#L90

**Comment by ogizanagi** [comment]
@yceruto : IIUC, the FrameworkExtensionTest does not build the FrameworkBundle, so compiler passes like the `LoggerPass` registered in the fwb are never processed.

Anyway, you moved it back to the MessengerPass, so this check is kind of required here as you can theoretically use the pass shipped within the component out of the Fwb usage, just with DI. So, let's keep this check.

**Review by ogizanagi — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml [hunk 5]
```diff
@@ -52,7 +52,7 @@
 
         <!-- Logging & Debug -->
         <service id="messenger.middleware.debug.logging" class="Symfony\Component\Messenger\Debug\LoggingMiddleware">
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml]: Another workaround could be creating from scratch this service definition in `MessengerPass`, only if debug and logger are enabled/exists, wdyt?

Comment (ogizanagi) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.xml]: We usually try as much as possible to declare all services in xml files so everything is in one place, favoring discovery/inspections and allowing to jump from classes to definition with the PhpStorm Symfony plugin.

**Review by nicolas-grekas — APPROVED** [review]

**Review by nicolas-grekas — DISMISSED** [review]
composer.json should be alpha ordered
could also validator+process work with 3.4? if yes, let's do the change here.

**Comment by yceruto** [comment]
Done! (failures are unrelated)

Status: Needs Review

**Review by sroze — APPROVED** [review]

**Comment by sroze** [comment]
Thank you @yceruto.
